src/common/datasource/src/file_format/parquet.rs

```diff
@@ -216,6 +221,31 @@ pub async fn stream_to_parquet(
     Ok(rows_written)
 }
 
+/// Customizes per-column properties.
+fn column_wise_config(
+    mut props: WriterPropertiesBuilder,
+    schema: SchemaRef,
+) -> WriterPropertiesBuilder {
+    // Disable dictionary for timestamp column.
+    if let Some(ts_col) = schema.timestamp_column() {
+        let path = ColumnPath::new(vec![ts_col.name.clone()]);
+        props = props.set_column_dictionary_enabled(path, false);
+    }
+
+    // Use delta binary packed encoding for int32/int64 columns.
+    for column in schema.column_schemas() {
+        let data_type = &column.data_type;
+        if data_type == &ConcreteDataType::int64_datatype()
+            || data_type == &ConcreteDataType::int32_datatype()
+        {
+            let path = ColumnPath::new(vec![column.name.clone()]);
+            props = props.set_column_encoding(path, Encoding::DELTA_BINARY_PACKED);
+        }
+    }
```

Comment:
I wonder if all int32/int64 should use delta binary packed encoding. Maybe we should only customize timestamp column.